src/components/ShareListDialog.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useRef } from 'react';
+import './ShareListDialog.css';
+
+const NO_OP = () => {};
+
+const DEFAULT_CLASSNAMES = {
+	content: 'Dialog--content',
+	form: 'Dialog--container',
+	root: 'Dialog',
+};
+
+/**
+ * @param {Omit<React.DialogHTMLAttributes<HTMLDialogElement>, 'className'> & {classNames: Partial<typeof DEFAULT_CLASSNAMES>}} props
+ */
+export function ShareListDialog({
+	children,
+	classNames,
```

Comment:
🔍 This `classNames` prop gives us the ability to pass classes to specific parts of the underlying component. Allowed properties are:
- `root`, for the `<dialog>` element
- `form`, for the `<form>` that wraps all of the dialog content
- `content`, for the innermost content region directly around the `children`